File: Problem 09.py
import numpy as np
import scipy.stats as stats
import statsmodels.api as sm

def calculate_correlation_and_regression(x, y):
  # Calculate correlation coefficient
  correlation = np.corrcoef(x, y)[0, 1]

  # Add a constant term for the intercept
  x = sm.add_constant(x)

  # Fit linear regression model
  model = sm.OLS(y, x).fit()

  # Extract regression coefficients
  intercept = model.params[0]
  slope = model.params[1]

  return correlation, intercept, slope

x = np.array([80, 100, 120, 140, 160, 180, 200, 220, 240, 260])
y = np.array([70, 65, 90, 95, 110, 115, 120, 140, 155, 150])
correlation, intercept, slope = calculate_correlation_and_regression(x, y)

File: Problem 09.r
calculate_correlation_and_regression <- function(x, y) {
  # Calculate correlation coefficient
  correlation <- cor(x, y)

  # Fit linear regression model
  model <- lm(y ~ x)

  # Extract regression coefficients
  intercept <- coef(model)[1]
  slope <- coef(model)[2]

  # Return correlation and regression equation
  list(correlation = correlation, intercept = intercept, slope = slope)
}

x <- c(80, 100, 120, 140, 160, 180, 200, 220, 240, 260)
y <- c(70, 65, 90, 95, 110, 115, 120, 140, 155, 150)

result <- calculate_correlation_and_regression(x, y)
correlation <- result$correlation
intercept <- result$intercept
slope <- result$slope

print(paste("Correlation coefficient:", correlation))
print(paste("Regression equation: Y = ", intercept, " + ", slope, " * X"))